### PR TITLE
[ZEPPELIN-4890]. Support to run note in isolated mode in rest api

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR is to support to run note in isolated mode in rest api, that means all the interpreter in this rest api is isolated and will be released after the note is finished. This is used for other third party that use rest api to integrate with zeppelin to run note, e.g. airflow use it to run ETL job.

### What type of PR is it?
[Improvement | Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-4890

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
This is a really useful feature!

Do you think it makes sense to have a Note revision option to isolated mode run?

For example, HEAD revision can be accidentally changed and external ETL jobs via REST API or Scheduled Note via embedded cron scheduler can fail or run unexpected code.

If external ETL job / cron scheduled job runs particular revision, users can change Note without worries.
What do you think?

**Comment by zjffdu** [comment]
Thanks @Leemoonsoo for the suggestion, that make sense to me, let me create a ticket for it and do it in a seprate PR.

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContextBuilder.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    this.inIsolatedMode = inIsolatedMode;
+    return this;
+  }
+
+  public ExecutionContextBuilder setStartTime(String startTime) {
+    this.startTime = startTime;
+    return this;
+  }
+
+  public ExecutionContext createExecutionContext() {
```
[the remark this answers is not in the document]
Comment (Reamer) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContextBuilder.java:55]: Sry I'm to late...

[the remark this answers is not in the document]
Comment (zjffdu) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContextBuilder.java:55]: Thanks for the review @Reamer , I use the template of intellij. Using build make sense to me, I can do it in another PR.

[the remark this answers is not in the document]
Comment (Reamer) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContextBuilder.java:55]: That would be great.

**Pull request closed without merging**